TesseractCoreTests/Assertions.swift

```diff
@@ -55,6 +56,19 @@ extension XCTestCase {
 			return failure("\(actual) is not equal to \(expected). " + message, file: file, line: line)
 		}
 	}
+    
+    func assertEqual<T: Equatable where T: Hashable>(@autoclosure expression1: () -> Graph<T>, @autoclosure _ expression2: () -> Graph<T>, _ message: String = "", _ file: String = __FILE__, _ line: UInt = __LINE__) -> Graph<T>? {
+        let graph1 = expression1()
+        let graph2 = expression2()
+        let lhsNodes = Set(graph1.nodes.values.array)
+        let rhsNodes = Set(graph2.nodes.values.array)
+        if (lhsNodes == rhsNodes && graph1.edges == graph2.edges) {
```

Comment:
There’s a bug here: duplicate node values are totally valid, so making sets out of the node values will potentially result in false positives.

Fortunately, since we’re constraining this to `T: Equatable`, we can use `==` over the graphs directly.